modules/core/src/test/java/org/apache/ignite/internal/ClusterMetricsSelfTest.java

```diff
@@ -95,6 +95,42 @@ public void testTaskExecution() {
         }
     }
 
+    public void testAllCachesNamesInCluster() throws InterruptedException {
+        Ignite g0 = grid(0);
+        Ignite g1 = grid(1);
+        Ignite g2 = grid(2);
+        Ignite g3 = grid(3);
+
+
+        g0.getOrCreateCache("Cache1");
+        g1.getOrCreateCache("Cache2");
+        g1.getOrCreateCache("Cache3");
+
+        Thread.sleep(20);
```

Comment:
When I run this unit test code at my desktop, I found this case would fail sporadic, from the log I can see the some of the node does not contains the cache, as the cache does not publish the the node. so here I let it sleep 20ms. with this line the test can pass always.